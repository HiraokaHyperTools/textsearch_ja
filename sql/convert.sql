\set ECHO all
SET client_encoding = utf8;

SELECT ja_normalize('０１２３４５６７８９0123456789');
SELECT ja_normalize('abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ');
SELECT ja_normalize('ａｂｃｄｅｆｇｈｉｊｋｌｍｎｏｐｑｒｓｔｕｖｗｘｙｚＡＢＣＤＥＦＧＨＩＪＫＬＭＮＯＰＱＲＳＴＵＶＷＸＹＺ');
SELECT ja_normalize('あいうえおアイウエオ');
SELECT ja_normalize(E'!"#$%&''()-=^~\\|@`[{;+:*]},<.>/?_');
SELECT ja_normalize('！”＃＄％＆’（）－＝＾￣￥｜＠‘［｛；＋：＊］｝，＜．＞／？＿');
SELECT ja_normalize('ｰｧｱｨｲｩｳｪｴｫｵｶｷｸｹｺｻｼｽｾｿﾀﾁｯﾂﾃﾄﾅﾆﾇﾈﾉﾊﾋﾌﾍﾎﾏﾐﾑﾒﾓｬﾔｭﾕｮﾖﾗﾘﾙﾚﾛﾜｦﾝ');
SELECT ja_normalize('ｶﾞｷﾞｸﾞｹﾞｺﾞｻﾞｼﾞｽﾞｾﾞｿﾞﾀﾞﾁﾞﾂﾞﾃﾞﾄﾞﾊﾞﾊﾟﾋﾞﾋﾟﾌﾞﾌﾟﾍﾞﾍﾟﾎﾞﾎﾟｳﾞ');
SELECT ja_normalize('ｶ゛ｷ゛ｸ゛ｹ゛ｺ゛ｻ゛ｼ゛ｽ゛ｾ゛ｿ゛ﾀ゛ﾁ゛ﾂ゛ﾃ゛ﾄ゛ﾊ゛ﾊ゜ﾋ゛ﾋ゜ﾌ゛ﾌ゜ﾍ゛ﾍ゜ﾎ゛ﾎ゜ｳ゛');
SELECT ja_normalize('カﾞキﾞクﾞケﾞコﾞサﾞシﾞスﾞセﾞソﾞタﾞチﾞツﾞテﾞトﾞハﾞハﾟヒﾞヒﾟフﾞフﾟヘﾞヘﾟホﾞホﾟウﾞ');
SELECT ja_normalize('カ゛キ゛ク゛ケ゛コ゛サ゛シ゛ス゛セ゛ソ゛タ゛チ゛ツ゛テ゛ト゛ハ゛ハ゜ヒ゛ヒ゜フ゛フ゜ヘ゛ヘ゜ホ゛ホ゜ウ゛');
SELECT ja_normalize('かﾞきﾞくﾞけﾞこﾞさﾞしﾞすﾞせﾞそﾞたﾞちﾞつﾞてﾞとﾞはﾞはﾟひﾞひﾟふﾞふﾟへﾞへﾟほﾞほﾟうﾞ');
SELECT ja_normalize('か゛き゛く゛け゛こ゛さ゛し゛す゛せ゛そ゛た゛ち゛つ゛て゛と゛は゛は゜ひ゛ひ゜ふ゛ふ゜へ゛へ゜ほ゛ほ゜う゛');

SELECT hiragana('あいうえおかきくけこさしすせそたちつてとなにぬねのはひふへほ');
SELECT hiragana('まみむめもやゐゆゑよらりるれろわをんぁぃぅぇぉっゃゅょゎ');
SELECT hiragana('がぎぐげござじずぜぞだぢづでどばびぶべぼぱぴぷぺぽ');
SELECT hiragana('アイウエオカキクケコサシスセソタチツテトナニヌネノハヒフヘホ');
SELECT hiragana('マミムメモヤヰユヱヨラリルレロワヲンァィゥェォッャュョヮ');
SELECT hiragana('ガギグゲゴザジズゼゾダヂヅデドバビブベボパピプペポヴ');
SELECT hiragana('ｱｲｳｴｵｶｷｸｹｺｻｼｽｾｿﾀﾁﾂﾃﾄﾅﾆﾇﾈﾉﾊﾋﾌﾍﾎ');
SELECT hiragana('ﾏﾐﾑﾒﾓﾔﾕﾖﾗﾘﾙﾚﾛﾜｦﾝｧｨｩｪｫｯｬｭｮ');
SELECT hiragana('ｶﾞｷﾞｸﾞｹﾞｺﾞｻﾞｼﾞｽﾞｾﾞｿﾞﾀﾞﾁﾞﾂﾞﾃﾞﾄﾞﾊﾞﾋﾞﾌﾞﾍﾞﾎﾞﾊﾟﾋﾟﾌﾟﾍﾟﾎﾟｳﾞ');

SELECT katakana('あいうえおかきくけこさしすせそたちつてとなにぬねのはひふへほ');
SELECT katakana('まみむめもやゐゆゑよらりるれろわをんぁぃぅぇぉっゃゅょゎ');
SELECT katakana('がぎぐげござじずぜぞだぢづでどばびぶべぼぱぴぷぺぽ');
SELECT katakana('アイウエオカキクケコサシスセソタチツテトナニヌネノハヒフヘホ');
SELECT katakana('マミムメモヤヰユヱヨラリルレロワヲンァィゥェォッャュョヮ');
SELECT katakana('ガギグゲゴザジズゼゾダヂヅデドバビブベボパピプペポヴ');
