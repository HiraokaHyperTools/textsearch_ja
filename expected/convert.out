\set ECHO all
SET client_encoding = utf8;
SELECT ja_normalize('０１２３４５６７８９0123456789');
     ja_normalize     
----------------------
 01234567890123456789
(1 row)

SELECT ja_normalize('abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ');
                     ja_normalize                     
------------------------------------------------------
 abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ
(1 row)

SELECT ja_normalize('ａｂｃｄｅｆｇｈｉｊｋｌｍｎｏｐｑｒｓｔｕｖｗｘｙｚＡＢＣＤＥＦＧＨＩＪＫＬＭＮＯＰＱＲＳＴＵＶＷＸＹＺ');
                     ja_normalize                     
------------------------------------------------------
 abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ
(1 row)

SELECT ja_normalize('あいうえおアイウエオ');
     ja_normalize     
----------------------
 あいうえおアイウエオ
(1 row)

SELECT ja_normalize(E'!"#$%&''()-=^~\\|@`[{;+:*]},<.>/?_');
           ja_normalize           
----------------------------------
 !"#$%&'()-=^~\|@`[{;+:*]},<.>/?_
(1 row)

SELECT ja_normalize('！”＃＄％＆’（）－＝＾￣￥｜＠‘［｛；＋：＊］｝，＜．＞／？＿');
           ja_normalize           
----------------------------------
 !"#$%&'()-=^~\|@`[{;+:*]},<.>/?_
(1 row)

SELECT ja_normalize('ｰｧｱｨｲｩｳｪｴｫｵｶｷｸｹｺｻｼｽｾｿﾀﾁｯﾂﾃﾄﾅﾆﾇﾈﾉﾊﾋﾌﾍﾎﾏﾐﾑﾒﾓｬﾔｭﾕｮﾖﾗﾘﾙﾚﾛﾜｦﾝ');
                                                   ja_normalize                                                   
------------------------------------------------------------------------------------------------------------------
 ーァアィイゥウェエォオカキクケコサシスセソタチッツテトナニヌネノハヒフヘホマミムメモャヤュユョヨラリルレロワヲン
(1 row)

SELECT ja_normalize('ｶﾞｷﾞｸﾞｹﾞｺﾞｻﾞｼﾞｽﾞｾﾞｿﾞﾀﾞﾁﾞﾂﾞﾃﾞﾄﾞﾊﾞﾊﾟﾋﾞﾋﾟﾌﾞﾌﾟﾍﾞﾍﾟﾎﾞﾎﾟｳﾞ');
                     ja_normalize                     
------------------------------------------------------
 ガギグゲゴザジズゼゾダヂヅデドバパビピブプベペボポヴ
(1 row)

SELECT ja_normalize('ｶ゛ｷ゛ｸ゛ｹ゛ｺ゛ｻ゛ｼ゛ｽ゛ｾ゛ｿ゛ﾀ゛ﾁ゛ﾂ゛ﾃ゛ﾄ゛ﾊ゛ﾊ゜ﾋ゛ﾋ゜ﾌ゛ﾌ゜ﾍ゛ﾍ゜ﾎ゛ﾎ゜ｳ゛');
                     ja_normalize                     
------------------------------------------------------
 ガギグゲゴザジズゼゾダヂヅデドバパビピブプベペボポヴ
(1 row)

SELECT ja_normalize('カﾞキﾞクﾞケﾞコﾞサﾞシﾞスﾞセﾞソﾞタﾞチﾞツﾞテﾞトﾞハﾞハﾟヒﾞヒﾟフﾞフﾟヘﾞヘﾟホﾞホﾟウﾞ');
                     ja_normalize                     
------------------------------------------------------
 ガギグゲゴザジズゼゾダヂヅデドバパビピブプベペボポヴ
(1 row)

SELECT ja_normalize('カ゛キ゛ク゛ケ゛コ゛サ゛シ゛ス゛セ゛ソ゛タ゛チ゛ツ゛テ゛ト゛ハ゛ハ゜ヒ゛ヒ゜フ゛フ゜ヘ゛ヘ゜ホ゛ホ゜ウ゛');
                     ja_normalize                     
------------------------------------------------------
 ガギグゲゴザジズゼゾダヂヅデドバパビピブプベペボポヴ
(1 row)

SELECT ja_normalize('かﾞきﾞくﾞけﾞこﾞさﾞしﾞすﾞせﾞそﾞたﾞちﾞつﾞてﾞとﾞはﾞはﾟひﾞひﾟふﾞふﾟへﾞへﾟほﾞほﾟうﾞ');
                     ja_normalize                     
------------------------------------------------------
 がぎぐげござじずぜぞだぢづでどばぱびぴぶぷべぺぼぽヴ
(1 row)

SELECT ja_normalize('か゛き゛く゛け゛こ゛さ゛し゛す゛せ゛そ゛た゛ち゛つ゛て゛と゛は゛は゜ひ゛ひ゜ふ゛ふ゜へ゛へ゜ほ゛ほ゜う゛');
                     ja_normalize                     
------------------------------------------------------
 がぎぐげござじずぜぞだぢづでどばぱびぴぶぷべぺぼぽヴ
(1 row)

SELECT hiragana('あいうえおかきくけこさしすせそたちつてとなにぬねのはひふへほ');
                           hiragana                           
--------------------------------------------------------------
 あいうえおかきくけこさしすせそたちつてとなにぬねのはひふへほ
(1 row)

SELECT hiragana('まみむめもやゐゆゑよらりるれろわをんぁぃぅぇぉっゃゅょゎ');
                         hiragana                         
----------------------------------------------------------
 まみむめもやゐゆゑよらりるれろわをんぁぃぅぇぉっゃゅょゎ
(1 row)

SELECT hiragana('がぎぐげござじずぜぞだぢづでどばびぶべぼぱぴぷぺぽ');
                      hiragana                      
----------------------------------------------------
 がぎぐげござじずぜぞだぢづでどばびぶべぼぱぴぷぺぽ
(1 row)

SELECT hiragana('アイウエオカキクケコサシスセソタチツテトナニヌネノハヒフヘホ');
                           hiragana                           
--------------------------------------------------------------
 あいうえおかきくけこさしすせそたちつてとなにぬねのはひふへほ
(1 row)

SELECT hiragana('マミムメモヤヰユヱヨラリルレロワヲンァィゥェォッャュョヮ');
                         hiragana                         
----------------------------------------------------------
 まみむめもやゐゆゑよらりるれろわをんぁぃぅぇぉっゃゅょゎ
(1 row)

SELECT hiragana('ガギグゲゴザジズゼゾダヂヅデドバビブベボパピプペポヴ');
                       hiragana                       
------------------------------------------------------
 がぎぐげござじずぜぞだぢづでどばびぶべぼぱぴぷぺぽヴ
(1 row)

SELECT hiragana('ｱｲｳｴｵｶｷｸｹｺｻｼｽｾｿﾀﾁﾂﾃﾄﾅﾆﾇﾈﾉﾊﾋﾌﾍﾎ');
                           hiragana                           
--------------------------------------------------------------
 あいうえおかきくけこさしすせそたちつてとなにぬねのはひふへほ
(1 row)

SELECT hiragana('ﾏﾐﾑﾒﾓﾔﾕﾖﾗﾘﾙﾚﾛﾜｦﾝｧｨｩｪｫｯｬｭｮ');
                      hiragana                      
----------------------------------------------------
 まみむめもやゆよらりるれろわをんぁぃぅぇぉっゃゅょ
(1 row)

SELECT hiragana('ｶﾞｷﾞｸﾞｹﾞｺﾞｻﾞｼﾞｽﾞｾﾞｿﾞﾀﾞﾁﾞﾂﾞﾃﾞﾄﾞﾊﾞﾋﾞﾌﾞﾍﾞﾎﾞﾊﾟﾋﾟﾌﾟﾍﾟﾎﾟｳﾞ');
                       hiragana                        
-------------------------------------------------------
 がぎぐげござじずぜぞだぢづでどばびぶべぼぱぴぷぺぽうﾞ
(1 row)

SELECT katakana('あいうえおかきくけこさしすせそたちつてとなにぬねのはひふへほ');
                           katakana                           
--------------------------------------------------------------
 アイウエオカキクケコサシスセソタチツテトナニヌネノハヒフヘホ
(1 row)

SELECT katakana('まみむめもやゐゆゑよらりるれろわをんぁぃぅぇぉっゃゅょゎ');
                         katakana                         
----------------------------------------------------------
 マミムメモヤヰユヱヨラリルレロワヲンァィゥェォッャュョヮ
(1 row)

SELECT katakana('がぎぐげござじずぜぞだぢづでどばびぶべぼぱぴぷぺぽ');
                      katakana                      
----------------------------------------------------
 ガギグゲゴザジズゼゾダヂヅデドバビブベボパピプペポ
(1 row)

SELECT katakana('アイウエオカキクケコサシスセソタチツテトナニヌネノハヒフヘホ');
                           katakana                           
--------------------------------------------------------------
 アイウエオカキクケコサシスセソタチツテトナニヌネノハヒフヘホ
(1 row)

SELECT katakana('マミムメモヤヰユヱヨラリルレロワヲンァィゥェォッャュョヮ');
                         katakana                         
----------------------------------------------------------
 マミムメモヤヰユヱヨラリルレロワヲンァィゥェォッャュョヮ
(1 row)

SELECT katakana('ガギグゲゴザジズゼゾダヂヅデドバビブベボパピプペポヴ');
                       katakana                       
------------------------------------------------------
 ガギグゲゴザジズゼゾダヂヅデドバビブベボパピプペポヴ
(1 row)

